### ISSUE TITLE
Q: build for (ARM 32bit) `armeabi` or `armeabiv7a` ?

### ISSUE THREAD

**Issue by jwmh** [thewierdnut/endless-mobile#26]
EDIT / UPDATE:
Just came across this, and added some comments there:
https://github.com/thewierdnut/endless-mobile/commit/9536ce3ebfe3683d0259e6fa63928a7951ecf83f

### Problem Description

Q: build for (ARM 32bit) `armeabi` or `armeabiv7a` ?

Currently,
FDroid tells me:
```
This app is incompatible with your device.
(requires arm64-v8a)
```

### Related Issue Links

Maybe helpful?

1. https://stackoverflow.com/questions/56824557/what-is-the-difference-between-armeabi-v7a-arm64-v8a-x86?answertab=modifieddesc#tab-top
2. https://stackoverflow.com/questions/23042175/what-is-armeabi-and-why-they-use-it?answertab=modifieddesc#tab-top
3. https://stackoverflow.com/questions/46453457/which-android-abis-cpu-architectures-do-i-need-to-serve
4. https://developer.android.com/ndk/guides/abis

### Desired Solution

I don't really understand this (the programming / compiling / building aspects, I mean).

I do understand the difference between different CPU architectures!

But, not how to compile & build for different Android devices.

The most I've ever done is carefully ensure I have all dependencies on Ubuntu met,  and then run 'make && make install' and then some other stuff and somehow it magically all works... in the case of .deb stuff being compiled locally on a debian system and installed therein..... But cross-compiling, if that's what this is.... I have zero clue.

So,
How might I do this on a Desktop (e.g. Winx64) for target: armeabi / armeabi v7a ?

### Alternative Approaches

... Or,
Can I compile / build this on Android itself?
as a non-root user,
on a stock, locked Android armeabi-v7a device?

I mean, can I take this source code (download as ZIP and unzip),  and somehow compile it inside of Termux into a workable .apk for my own device
???

### Additional Context

... Or can it only be compiled on a device with root access?   (e.g. a desktop/laptop, or else a rooted droid?)

I apologise.
I realise this is *SO* not the place to be asking these sort of n00b questions.
Still...
... maybe others here would be interested, too?

Feel free to disregard this btw.
I know this is super low priority.
But maybe a good learning opportunity for newbie compilers / bug reporters?

### edit/updates:
Reading over those four (4) links above has given me a general starting sense.
It looks like there's more for me to learn in the 4th link:
4. https://developer.android.com/ndk/guides/abis
...
Except...
`"Note: Historically the NDK supported ARMv5 (armeabi), and 32-bit and 64-bit MIPS, but support for these ABIs was removed in NDK r17."`

But then.
reading from the top...
and scrolling down the ENTIRE page.
More useful info on how to use Gradle and others.
https://developer.android.com/ndk/guides/abis#gc

Options are given for 
- Gradle
- ndk-build
- CMake

For brevity,
I'm only including the excerpt for ndk-build, here (but all three are available w code examples at the above link; they seem fairly similar and brief!).

"
> ndk-build builds for **all non-deprecated ABIs by default**. You can target a specific ABIs by setting APP_ABI in your [Application.mk](https://developer.android.com/ndk/guides/application_mk) file. The following snippet shows a few examples of using APP_ABI:
>
```
APP_ABI := arm64-v8a  # Target only arm64-v8a
APP_ABI := all  # Target **all** ABIs, **_including those that are deprecated._**
APP_ABI := armeabi-v7a x86_64  # Target only armeabi-v7a and x86_64.
```
> For more information on the values you can specify for APP_ABI, see [Application.mk](https://developer.android.com/ndk/guides/application_mk).

The default behavior of the build system is to include the binaries for each ABI in a single APK, also known as a [fat APK](https://en.wikipedia.org/wiki/Fat_binary).
 A fat APK is significantly larger than one containing only the binaries for a single ABI; the tradeoff is gaining wider compatibility, but at the expense of a larger APK.
 It is strongly recommended that you take advantage of either [App Bundles](https://developer.android.com/guide/app-bundle) or [APK Splits](https://developer.android.com/studio/build/configure-apk-splits) to reduce the size of your APKs while still maintaining maximum device compatibility.

"

EDIT / UPDATE:
Just came across this, and added some comments there:
https://github.com/thewierdnut/endless-mobile/commit/9536ce3ebfe3683d0259e6fa63928a7951ecf83f

**Comment by jwmh** [comment]
EDIT / UPDATE:
Just came across this, and added some comments there:
https://github.com/thewierdnut/endless-mobile/commit/9536ce3ebfe3683d0259e6fa63928a7951ecf83f

**Comment by thewierdnut** [comment]
F-droid does not have the armv7a enabled build yet. 

What android version/device do you have? I managed to dig up an old marshmallow tablet with a dead backlight, and without being able to tell what is actually on the screen, I can tell that the latest code runs, but not much else.

**Comment by thewierdnut** [comment]
I have this working on an old armabi-v7a device, although I had to implement a "reduced graphics mode" to get it to fit into memory.

**Comment by thewierdnut** [comment]
fixed by 8d8d6710
